### PR TITLE
fix(notifier): fix fetching notifications by batch

### PR DESCRIPTION
**Pull request by almostinf**
# Fix fetching notifications by batch

After the merge PR #946 it will be broken to receive notifications with a limit, this PR solves that problem

I replaced the `IgnoredNotification` state with `ResendNotification` so that all notifications whose trigger or metric is on Maintenance are resaved in the database, this is to avoid creating a large queue of such notifications with an unupdated Timestamp in the database because they would interfere with fetching notifications with a limit. Now all such notifications will be deleted, and their copies with updated Timestamp on `resave_time` will be stored in the database

### CODE REVIEW

File: database/redis/notification.go [hunk 1]
```diff
@@ -146,8 +139,16 @@ func (connector *DbConnector) GetDelayedTimeInSeconds() int64 {
 	return int64(connector.notification.DelayedTime.Seconds())
 }
 
+// GetResaveTimeInSeconds returns the time to reschedule notifications to the future in seconds
+func (connector *DbConnector) GetResaveTimeInSeconds() int64 {
```
Comment (Tetrergeru) [inline database/redis/notification.go:143]: А мы её снаружи испольщуем? Нельзя заприватить?

File: database/redis/notification.go [hunk 2]
```diff
@@ -511,7 +514,27 @@ func (connector *DbConnector) AddNotifications(notifications []*moira.ScheduledN
...
 
 	if err != nil {
-		return fmt.Errorf("failed to EXEC: %s", err.Error())
+		return fmt.Errorf("failed to EXEC: %w", err)
+	}
+
+	return nil
+}
+
+// saveNotifications save notifications with the passed pipe and context
```
Comment (Tetrergeru) [inline database/redis/notification.go:523]: Зачем годок к приватной функции?

File: database/redis/notification_test.go [hunk 3]
```diff
@@ -1001,6 +1014,9 @@ func TestFetchNotificationsDo(t *testing.T) {
 			dataBase.SetTriggerCheckMaintenance("test2", map[string]int64{"test2": time.Now().Add(time.Hour).Unix()}, nil, "test", 100) //nolint
 			defer dataBase.SetTriggerCheckMaintenance("test2", map[string]int64{"test2": 0}, nil, "test", 100)                          //nolint
 
+			updatedNotificationNew3 := notificationNew3
+			updatedNotificationNew3.Timestamp += dataBase.GetResaveTimeInSeconds()
```
Comment (Tetrergeru) [inline database/redis/notification_test.go:1018]: Может сразу переименуем `dataBase` -> `database`? Мне очень глаза режет :sweat_smile:

File: database/redis/notification.go (hunk 1 above)
Comment (almostinf) [reply database/redis/notification.go]: Заприватил

File: database/redis/notification.go (hunk 2 above)
Comment (almostinf) [reply database/redis/notification.go]: Убрал

File: database/redis/notification_test.go (hunk 3 above)
Comment (almostinf) [reply database/redis/notification_test.go]: Переименовал)

**Pull request closed without merging**